crates/store/re_video/src/decode/mod.rs

```diff
@@ -198,23 +198,49 @@ pub struct Chunk {
     pub duration: Time,
 }
 
+/// Data for a decoded frame on native targets.
 #[cfg(not(target_arch = "wasm32"))]
-pub type FrameData = Vec<u8>;
-
-#[cfg(target_arch = "wasm32")]
-pub type FrameData = webcodecs::WebVideoFrame;
-
-/// One decoded video frame.
-pub struct Frame {
-    pub data: FrameData,
+pub struct FrameContent {
+    pub data: Vec<u8>,
     pub width: u32,
     pub height: u32,
     pub format: PixelFormat,
+}
+
+/// Data for a decoded frame on the web.
+#[cfg(target_arch = "wasm32")]
+pub type FrameContent = webcodecs::WebVideoFrame;
+
+/// Meta information about a decoded video frame, as reported by the decoder.
+#[derive(Debug, Clone)]
+pub struct FrameInfo {
     pub presentation_timestamp: Time,
     pub duration: Time,
 }
 
-/// Pixel format/layout used by [`Frame::data`].
+impl Default for FrameInfo {
+    fn default() -> Self {
+        Self {
+            presentation_timestamp: Time::MAX,
+            duration: Time::MAX,
```

Comment:
Should these be optional instead? Or not have a `default()` ?

Reply:
the way we implement things we have to force presentation timestamp and I didn't mind sticking with this since that's what _most_ decoders come back with anyways. Duration is more of an issue, but fine for now
setting these to max is consistent with how we handled "lack of frame" so far.
I'll note that down in the code!